installer/pkg/workflow/install.go

```diff
@@ -108,7 +108,7 @@ func installJoinMastersStep(m *metadata) error {
 }
 
 func installJoinWorkersStep(m *metadata) error {
-	return runInstallStep(m, joinWorkersStep)
+	return nil
```

Comment:
If worker-joining is no longer going to be supported (which makes sense with a worker machine set), then I'd rather remove `installJoinWorkersStep` entirely instead of keeping it as a no-op.

Reply:
I'm happy to do that if you want, how ever I'd suggest to keep it for the sake of a smaller PR. Then in a follow up I will remove the step along with all deprecated terraform code

Reply:
Just throw all of that into this PR - that's why there are separate commits. It is logically related.